### PR TITLE
JAMES-3431 MockSMTPServer: Capture MAIL FROM & RCPT TO ESMTP parameters

### PR DESCRIPTION
**Pull request by chibenwa**

### CODE REVIEW

**Review by Arsnael — APPROVED** [review]

**Review by LanKhuat — APPROVED** [review]

**Comment by Arsnael** [comment]
Rebased on master

**Comment by chibenwa** [comment]
`java.lang.IllegalStateException: no known node for /home/jenkins/build/workspace/1142`

test this please

**Comment by Arsnael** [comment]
```
[bb0fa3f3b537a66df313e9799612841b6b7ca7f9] [ERROR] Failures: 
[bb0fa3f3b537a66df313e9799612841b6b7ca7f9] [ERROR]   MailboxChangeTest.shouldMatchBeanContract:40 EqualsVerifier found a problem in class MailboxChange.
[bb0fa3f3b537a66df313e9799612841b6b7ca7f9] -> Significant fields: equals does not use isCountChange, or it is stateless.
[bb0fa3f3b537a66df313e9799612841b6b7ca7f9] 
[bb0fa3f3b537a66df313e9799612841b6b7ca7f9] For more information, go to: http://www.jqno.nl/equalsverifier/errormessages
[bb0fa3f3b537a66df313e9799612841b6b7ca7f9] [ERROR]   MailboxChangeTest.shouldThrowOnNullAccountId:45 
[bb0fa3f3b537a66df313e9799612841b6b7ca7f9] Expecting code to raise a throwable.
[bb0fa3f3b537a66df313e9799612841b6b7ca7f9] [ERROR]   MailboxChangeTest.shouldThrowOnNullDate:65 
[bb0fa3f3b537a66df313e9799612841b6b7ca7f9] Expecting code to raise a throwable.
[bb0fa3f3b537a66df313e9799612841b6b7ca7f9] [ERROR]   MailboxChangeTest.shouldThrowOnNullState:55 
[bb0fa3f3b537a66df313e9799612841b6b7ca7f9] Expecting code to raise a throwable.
```

**Comment by chibenwa** [comment]
test this please

**Comment by chibenwa** [comment]
`no known node for /home/jenkins/build/workspace/1158` test this please

**Comment by Arsnael** [comment]
test this please

**Comment by Arsnael** [comment]
Merged

File: server/mailet/mock-smtp-server/src/main/java/org/apache/james/mock/smtp/server/ExtendedMailFromCommand.java [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+
+                try {
+                    sess.startMailTransaction();
+                    MockMessageHandler messageHandler = (MockMessageHandler) sess.getMessageHandler();
+                    messageHandler.from(emailAddress, Mail.Parameter.fromArgLine(args));
+                    sess.setDeclaredMessageSize(size);
+                    sess.setHasMailFrom(true);
+                    sess.sendResponse("250 Ok");
+                } catch (DropConnectionException var9) {
+                    throw var9;
```
Comment (rouazana) [inline server/mailet/mock-smtp-server/src/main/java/org/apache/james/mock/smtp/server/ExtendedMailFromCommand.java:77]: this has been reviewed?

Comment (Arsnael) [reply server/mailet/mock-smtp-server/src/main/java/org/apache/james/mock/smtp/server/ExtendedMailFromCommand.java:77]: Yes but likely got missed

Comment (Arsnael) [reply server/mailet/mock-smtp-server/src/main/java/org/apache/james/mock/smtp/server/ExtendedMailFromCommand.java:77]: would that answer your concern? https://github.com/linagora/james-project/pull/4198

**Pull request closed without merging**